torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -130,6 +139,86 @@ def _lower_weighted_ref_module(model: QuantizedGraphModule) -> QuantizedGraphMod
         model.recompile()
     return model
 
+def _lower_weighted_ref_functional(model: QuantizedGraphModule) -> QuantizedGraphModule:
+    """
+    Traverse the graph and replace functional reference patterns with their quantized versions.
+    """
+    for ref_func, (q_func, q_relu_func) in LOWER_FUNCTIONAL_MAP.items():
+        configurations = itertools.product(
+            (False, True),  # is_relu: whether ref_func is wrapped in a relu op
+            (False, True),  # has_bias: whether bias is passed as an extra argument to ref_func
+        )
+        for is_relu, has_bias in configurations:
+            if is_relu and q_relu_func is None:
+                continue
+
+            # Set up match pattern: (dequantize - [relu_op - ] func_op - quantize)
+            # Func args: (dequantized inputs, dequantized weights[, bias])
+            # Quantize args: (func, scale, zp, dtype)
+            func_pattern: Tuple[Any, ...] = (ref_func, "dequantize", "dequantize")
+            if has_bias:
+                func_pattern = tuple(list(func_pattern) + [MatchAllNode])
+            if is_relu:
+                func_pattern = (F.relu, func_pattern)
+            pattern = (torch.quantize_per_tensor, func_pattern, MatchAllNode, MatchAllNode, MatchAllNode)
```

Comment:
can we just write down the patterns?

Reply:
I mean we can write down the `func_pattern` part, instead of generating it

Reply:
Ok, since there are 4 configurations do you mean having something like this:
```
if has_bias and is_relu:
    func_pattern = ...
elif has_bias and not is_relu:
    func_pattern = ...
elif not has_bias and is_relu:
    func_pattern = ...
else:
    func_pattern = ...
```

Reply:
can we have something similar to https://github.com/pytorch/pytorch/pull/72953/files

Reply:
we can generate the easy ones with simple code, and for the ones that we can't easily generate, we can write down the pattern directly

Reply:
I wasn't able to follow https://github.com/pytorch/pytorch/pull/72953 exactly because here we need the pattern to match the configuration (e.g. `is_relu` + `has_bias`) in the loop. I did write out the `func_pattern` part more explicitly